inc/ThirdParty/Hostings/Godaddy.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\ThirdParty\ReturnTypesTrait;
+
+class Godaddy implements Subscriber_Interface {
+	use ReturnTypesTrait;
+
+	/**
+	 * Godaddy vip url
+	 *
+	 * @var bool|string
+	 */
+	private $vip_url;
+
+	/**
+	 * Godaddy constructor.
+	 *
+	 * @param bool|string $vip_url Godaddy vip url.
+	 */
+	public function __construct( $vip_url = false ) {
+		$this->vip_url = method_exists( '\WPaas\Plugin', 'vip' ) ? \WPaas\Plugin::vip() : $vip_url;
+	}
```

Comment:
I'm not sure about the point of passing the `$vip_url` as a parameter of the constructor. When we instantiate the class, we don't pass anything to it, so the value will never be different than `false` or the value from the `vip()` method?

Reply:
We did that to make the code testable so you may notice that the test is passing the `vip_url` but in the main code we will keep it with the default value of false and it will call this VIP method to get the vip_url that will be used here

https://github.com/wp-media/wp-rocket/blob/935d7362a6f93523e42cbc25f9d504c605be2a4b/inc/ThirdParty/Hostings/Godaddy.php#L141

The old code was bailing out if it didn't find the `vip` method here:
https://github.com/wp-media/wp-rocket/blob/cf8b0b647456783d62d9f180b8e71643fe75d763/inc/3rd-party/hosting/godaddy.php#L104-L106

so to create unit tests for that we need to create a mock class (Stub) as we did with integration tests here:

https://github.com/wp-media/wp-rocket/blob/935d7362a6f93523e42cbc25f9d504c605be2a4b/tests/Fixtures/inc/ThirdParty/Hostings/Godaddy/Plugin.php

but if we did that, this will affect the other tests as we require this class so always GoDaddy will be detected as the current host, and no way to change it.

so we have two options here:-

1. Don't add unit tests for GoDaddy methods (just the integration tests) and remove this vip_url argument.
2. keep unit tests and keep this argument.

What do you think?

Reply:
Ok I understand the need now 👍🏼 

What about having a setter method instead? `set_vip_url()`